### DISCUSSION TITLE
Unable to get authentication token

### DISCUSSION THREAD

**Discussion by andreiv3103** [CrowdStrike/psfalcon#218] [General]
Hello
I have the following error when trying to get the auth token:
![image](https://user-images.githubusercontent.com/2743608/175958340-7c9f37e2-73ff-47bd-9cad-da8ab5d1c12e.png)
Any idea why is this happening?
If I remove the Hostname parameter, it works.

But I need the hostname paramter because I need to authenticate againts different API URLs.

**Comment by andreiv3103** [comment]
Answering my own issue.
This happend because I ran the command in a session where a token was previously created.
And that seems to cause the error above. So, before requesting a new token, the previous one must be cleared.

**Reply by bk-cs** [reply]
You shouldn't need to revoke a previous token in order to get a new token. Are you manually importing the PSFalcon module before attempting the token request?

**Reply by andreiv3103** [reply]
No, the module is imported automatically.

**Reply by andreiv3103** [reply]
When I revoke the token before trying to get it from a different URL it works.
If I don't revoke, it fails with the error above when trying to get the token from the second url.

**Reply by bk-cs** [reply]
I recommend doing a manual import of the module before running commands. PowerShell doesn't always properly import the underlying `[ApiClient]` object that PSFalcon uses to make requests. When it's not properly imported, there can be errors related to parsing URLs.

**Reply by andreiv3103** [reply]
Yes, that was probably the issue.
I am actually using the cmdlets in a script.
There I am using the "using module PSFalcon" statement and, in the script, it works just fine. I did add the token revocation command between the 2 requests for token, just to be on the safe side. But it works.
The error I got above appeared while I was manually testing in the Windows Terminal PS console.
